### DISCUSSION TITLE
brew update/upgrade/install floods terminal with garbage on Linux

### DISCUSSION THREAD

**Discussion by AdrianGroty** [Homebrew/discussions#5644] [Linux] [answered]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.24
ORIGIN: https://github.com/Homebrew/brew
HEAD: 916044581862c32fc2365e8e9ff0b1507a98925e
Last commit: 7 days ago
Core tap JSON: 29 Sep 12:37 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0.0
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 3.3.5 => /snap/ruby/369/bin/ruby
CPU: dodeca-core 64-bit zen3
Clang: N/A
Git: 2.34.1 => /bin/git
Curl: 7.81.0 => /bin/curl
Kernel: Linux 6.8.0-45-generic x86_64 GNU/Linux
OS: Ubuntu 22.04.5 LTS (jammy)
Host glibc: 2.35
/usr/bin/gcc: 11.4.0
/usr/bin/ruby: N/A
glibc: N/A
gcc@11: N/A
gcc: 14.2.0
xorg: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Terminal floods with progress bar characters on brew update/upgrade/install. I have included an asciinema recording.

https://asciinema.org/a/tUTt2DH5KcBZDTHZuI1ci8i53

I have read of an issue with the progress bar taking up extra lines in terminals narrower than 80 columns, but the screenshots in those cases appear different, and my issue happens no matter how many columns are available.

https://github.com/Homebrew/brew/issues/981
https://github.com/Homebrew/brew/pull/1002

Any idea what gives?

**Comment by carlocab** [comment]
Might be `/snap/ruby/369/bin/ruby`, or some interaction with it and how `brew` and its subprocesses gather information about your terminal.

Can you reproduce it with a `ruby` you've installed through `rbenv`?

In a pinch, you can try passing `-q` or `--quiet`.

**Reply by AdrianGroty** [reply]
> Can you reproduce it with a `ruby` you've installed through `rbenv`?

Thank you for the quick reply!

Forgive my ignorance, but I am not entirely sure what a `ruby` is in this context or how I would go about reproducing this behavior with one.

**Reply by carlocab** [reply, marked as answer]
Homebrew is built using [Ruby](https://www.ruby-lang.org/en/), so it needs a `ruby` binary to run. Yours seems to be found at `/snap/ruby/369/bin/ruby`.

I suggest trying:
```
brew install --quiet rbenv openssl@3
eval "$(rbenv init -)"
export RUBY_CONFIGURE_OPTS="--with-openssl-dir=$(brew --prefix openssl@3)"
rbenv install 3.3.5
```
Once that's done (it may take a while), the output of `ruby --version` and `which ruby` should look something like (sans the `arm64-darwin23` part)
```
❯ ruby --version
ruby 3.3.5 (2024-09-03 revision ef084cc8f4) [arm64-darwin23]

❯ which ruby
/path/to/home/.rbenv/shims/ruby
```

Once you've done that, paste the output of `brew config` here and then try `brew install` again.

**Reply by AdrianGroty** [reply]
This seems to have done the trick. Thanks!

I figure the output of brew config probably isn't important now, but in case it is:

HOMEBREW_VERSION: 4.4.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: 84c31175f11860129a9aaed40a13c549625e2db1
Last commit: 19 hours ago
Core tap JSON: 02 Oct 06:53 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0.0
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 3.3.5 => /home/annehiiro/.rbenv/versions/3.3.5/bin/ruby
CPU: dodeca-core 64-bit zen3
Clang: N/A
Git: 2.34.1 => /bin/git
Curl: 7.81.0 => /bin/curl
Kernel: Linux 6.8.0-45-generic x86_64 GNU/Linux
OS: Ubuntu 22.04.5 LTS (jammy)
Host glibc: 2.35
/usr/bin/gcc: 11.4.0
/usr/bin/ruby: 3.0.2
glibc: N/A
gcc@11: N/A
gcc: 14.2.0
xorg: N/A

**Reply by carlocab** [reply]
Great. You'll want to add
```
eval "$(rbenv init -)"
```
to your shell startup scripts (e.g. `.bashrc`, `.zshrc`, `.bash_profile`, or similar) to make sure that `/home/annehiiro/.rbenv/shims` is in your `PATH` for `brew` to find it. You may need to do it after doing `brew shellenv` (if you use it), but before any other usage of `brew`.
